rt: use updateReply everywhere

corlewis opened the pull request:
This changes all of the remaining `getReply; setReply` patterns to use `updateReply`, except for `replyPop` which is left for https://github.com/seL4/l4v/pull/185.

mktnk3 commented on spec/haskell/src/SEL4/Object/Reply.lhs:
```diff
@@ -28,6 +28,11 @@ This module specifies the behavior of reply objects.
 
 \end{impdetails}
 
+> getReply :: PPtr Reply -> Kernel Reply
+> getReply rptr = getObject rptr
+
+> setReply :: PPtr Reply -> Reply -> Kernel ()
+> setReply rptr r = setObject rptr r
```
Do we need to move this?

corlewis replied:
Yeah, it's not needed but it made sense to me to have them next to `updateReply`, since that's now the main place that they're used. I'm happy to undo it if you'd prefer not.

mktnk3 replied:
Ah, I see. That makes sense! No need to move, happy for them to stay here.

mktnk3 reviewed (approved):
#185 has the same spec change for `replyRemove`, but in this case, the rebase should work cleanly, hopefully. There’s some overlap in `replyRemove_valid_objs`, but I suppose it won’t be too bad. Please keep in mind that both Victor and myself are working on top of #185 at the moment.

corlewis commented:
> 
> 
> #185 has the same spec change for `replyRemove`, but in this case, the rebase should work cleanly, hopefully. There’s some overlap in `replyRemove_valid_objs'`, but I suppose it won’t be too bad. Please keep in mind that both Victor and myself are working on top of #185 at the moment.

Sorry, I missed that #185 also updated `replyRemove`. I've removed it from this PR, which now just has the SchedContext changes I was originally thinking about.
